File: qt/aqt/buildinfo.py
# Copyright: Ankitects Pty Ltd and contributors
# License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html

import os

_buildinfo = {}
for line in open(os.path.join(os.path.dirname(__file__), "buildinfo.txt")).readlines():
    elems = line.split()
    if len(elems) == 2:
        k, v = elems
        _buildinfo[k] = v

buildhash=_buildinfo["STABLE_BUILDHASH"]
version=_buildinfo["STABLE_VERSION"]

File: mobile/ios/BrainliftMobile/App/BuildInfo.swift
// Copyright: Ankitects Pty Ltd and contributors
// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html

import Foundation

enum BuildInfo {
    static let sourceRevision = String(cString: anki_backend_source_revision())
    static let bundleSourceRevision =
        Bundle.main.object(forInfoDictionaryKey: "AnkiBridgeSourceRevision") as? String
    static let identityIsConsistent = bundleSourceRevision == sourceRevision
    static let shortSourceRevision =
        String(sourceRevision.prefix(9)) + (sourceRevision.hasSuffix("-dirty") ? "-dirty" : "")
    static let displaySourceRevision =
        identityIsConsistent ? shortSourceRevision : "\(shortSourceRevision) (bundle mismatch)"
}
